query/src/plan/stringset.rs

```diff
@@ -0,0 +1,257 @@
+use std::sync::Arc;
+
+use arrow_deps::{datafusion::logical_plan::LogicalPlan, util::str_iter_to_batch};
+use data_types::TABLE_NAMES_COLUMN_NAME;
+
+use crate::{
+    exec::stringset::{StringSet, StringSetRef},
+    util::make_scan_plan,
+};
+
+use snafu::{ResultExt, Snafu};
+
+#[derive(Debug, Snafu)]
+pub enum Error {
+    #[snafu(display("Internal error converting to arrow: {}", source))]
+    InternalConvertingToArrow {
+        source: arrow_deps::arrow::error::ArrowError,
+    },
+
+    #[snafu(display("Internal creating a plan for stringset: {}", source))]
+    InternalPlanningStringSet {
+        source: arrow_deps::datafusion::error::DataFusionError,
+    },
+}
+
+pub type Result<T, E = Error> = std::result::Result<T, E>;
+
+/// A plan which produces a logical set of Strings (e.g. tag
+/// values). This includes variants with pre-calculated results as
+/// well a variant that runs a full on DataFusion plan.
+#[derive(Debug)]
+pub enum StringSetPlan {
+    /// The plan errored and we are delaying reporting the results
+    KnownError(Box<dyn std::error::Error + Send + Sync + 'static>),
+
+    /// The results are known from metadta only without having to run
+    /// an actual datafusion plan
+    KnownOk(StringSetRef),
+
+    /// A DataFusion plan(s) to execute. Each plan must produce
+    /// RecordBatches with exactly one String column, though the
+    /// the values produced by the plan may be repeated
```

Comment:
![(PNG Image, 390 × 291 pixels)](https://user-images.githubusercontent.com/193874/107089940-768f6e00-67cd-11eb-8bd4-7dabd8710f03.png)

You can't get me with that trick!

```suggestion
    /// A DataFusion plan(s) to execute. Each plan must produce
    /// RecordBatches with exactly one String column, though the
    /// values produced by the plan may be repeated
```

Reply:
wow what 👁️ !